docs/getting-started/quickstart-programming.md

```diff
@@ -79,178 +62,231 @@ exports.up = function(knex) {
+  //Omitted for brevity: @GetApi('/greetings') //read greetings from database
+  
   @Communicator()
-  static async SendGreetingEmail(ctxt: CommunicatorContext, friend: string, content: string) {
-    ctxt.logger.info(`Sending email "${content}" to ${friend}...`);
-    // Code omitted for simplicity
-    ctxt.logger.info("Email sent!");
+  static async SignGuestbook(ctxt: CommunicatorContext, name: string) {
+    const response = await fetch('https://demo-guestbook.cloud.dbos.dev/record_greeting', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json'},
+      body: JSON.stringify({ 'key': process.env.GUESTBOOK_KEY, 'name': name})
+    });
+    const responseStr = JSON.stringify(await response.json());
+    if (!response.ok) {
+      throw new DBOSResponseError(responseStr);
+    }
+    ctxt.logger.info(`>>> STEP 1: Signed the Guestbook: ${responseStr}`);
   }
 
   @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
-    ctxt.logger.info(`Greeting to ${friend} recorded in the database!`);
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, gr: GreetingRecord) {
+    await ctxt.client('greetings').insert(gr);
+    ctxt.logger.info(`>>> STEP 2: Greeting to ${gr.name} recorded in the database!`);
   }
 
-  @GetApi("/greeting/:friend")
+  @GetApi('/greeting/:friend')
   static async Greeting(ctxt: HandlerContext, friend: string) {
     const noteContent = `Thank you for being awesome, ${friend}!`;
-    await ctxt.invoke(Greetings).SendGreetingEmail(friend, noteContent);
-    await ctxt.invoke(Greetings).InsertGreeting(friend, noteContent);
+    await ctxt.invoke(Greetings).SignGuestbook(friend);
+    await ctxt.invoke(Greetings).InsertGreeting(
+      { name: friend, note: noteContent }
+    );
     return noteContent;
   }
 }
 ```
 
-The key elements of this code are:
-- We use the [`@Communicator`](../api-reference/decorators#transaction) decorator to define a [communicator function](../tutorials/communicator-tutorial.md) (`SendGreetingEmail`) to access a third-party email service.
-- We add a line to `Greeting` invoking `SendGreetingEmail`.
+Note the new `@Communicator` function called `SignGuestbook` that uses a `fetch` call to send the request. If this communicator throws an error, it will be automatically retried up to 3 times with exponential backoff. This is configurable [here](https://docs.dbos.dev/tutorials/communicator-tutorial#configurable-retries). Stop your app with CTRL+C, rebuild with `npm run build` and start your application with `npx dbos start`. Make a few visits to the greeting URL in your browser, i.e. http://localhost:3000/greeting/Mike. With every new visit, the app should now print first that it's recorded the greeting in the guestbook, then that it's recorded the greeting in the database.
```

Comment:
```suggestion
We add a new `@Communicator` function called `SignGuestbook` that uses a `fetch` to send an HTTP POST request recording each greeting in the guestbook. If this communicator throws an error, it will be automatically retried up to 3 times with exponential backoff. This is configurable [here](https://docs.dbos.dev/tutorials/communicator-tutorial#configurable-retries). Stop your app with CTRL+C, rebuild with `npm run build` and start your application with `npx dbos start`. Make a few visits to the greeting URL in your browser, i.e. http://localhost:3000/greeting/Mike. With every new visit, the app should now print first that it's recorded the greeting in the guestbook, then that it's recorded the greeting in the database.
```